src/data_generator/connectivity.py

```diff
@@ -0,0 +1,28 @@
+import networkx as nx
+
+
+def adj_to_networkX(adj_list: dict[str: list[str]]) -> nx.Graph:
+    """
+    Converts our custom adjacency list (sampled from OSM) formatted graph to networkX graph.
+    OSM - open street map.
+    :param adj_list: adjacency-list represented graph,
+                     ex.: {'R2': ['D2', ...], ...}.
+    :return: graph in networkX format.
+    """
+    G = nx.Graph()
+    for node1, adj in adj_list.items():
+        for node2 in adj:
+            G.add_edge(node1, node2)
+    return G
+
+
+def is_connected(adj_list) -> bool:
+    """
+    Checks if our custom adjacency list (sampled from OSM) is connected.
+    OSM - open street map.
+    :param adj_list: adjacency-list represented graph,
+                     ex.: {'R2': ['D2', ...], ...}.
+    :return: whether or not the graph is connected.
+    """
+    G = adj_to_networkX(adj_list)
```

Comment:
Была такая мысль, но - "NetworkX provides basic functionality for visualizing graphs, but its main goal is to enable graph analysis rather than perform graph visualization. In the future, graph visualization functionality may be removed from NetworkX or only available as an add-on package. Proper graph visualization is hard, and we highly recommend that people visualize their graphs with tools dedicated to that task. Notable examples of dedicated and fully-featured graph visualization tools are [Cytoscape](http://www.cytoscape.org/), [Gephi](https://gephi.org/), [Graphviz](http://www.graphviz.org/)" - https://networkx.org/documentation/stable/reference/drawing.html